GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/GasStorePageController.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using UnityEngine;
+using UnityEngine.UI;
+
+// controller for the gas store page
+public class GasStorePageController : MonoBehaviour
+{
+    public Text gasPrice;
+    public Text panelGasPrice;
+    public GameObject panelFillGas;
+    public GameObject gasLevelImageObj;
+    public GameObject cannotAffordWarning;
+    public GameObject car; // need to get gas class from car
+
+    private double _currentCost;
+    private Gas _gas;
+    private Image _gasLevelImage;
+    private GameData _gameData;
+
+    private void Awake()
+    {
+        _gas = car.GetComponent<Gas>();
+        _gasLevelImage = gasLevelImageObj.GetComponent<Image>();
+        _gameData = FindObjectOfType<GameManager>().GetGameData();
+    }
+
+    private void OnEnable()
+    {
+        RefreshGasStorePage();
+    }
+
+    // update the gas price and refresh the page when get into the gas store page
+    private void RefreshGasStorePage()
+    {
+        _currentCost = Math.Ceiling(_gas.GetFullGasLevel() - _gas.GetGasLevel());
+        gasPrice.text = "* " + _currentCost;
+        panelGasPrice.text = "Would you like to fill the gas tank with  " + _currentCost + "  coins";
+        _gas.SetGasLevelHelper(_gasLevelImage, gasLevelImageObj);
+        panelFillGas.SetActive(false);
+        cannotAffordWarning.SetActive(false);
+    }
+
+    // listener for fill gas button.
+    public void FillGasOnClick()
```

Comment:
how does this method get called? Through Unity object settings?

A better way to name the listener will be OnSomethingHappened()

Reply:
got it!